server.js

```diff
@@ -1,29 +1,78 @@
+/* eslint-disable max-len */
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+// eslint-disable-next-line max-len
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String, 
+    required: true,    
+    minlength: 5,
+    maxlength: 140
+  },   
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  CreatedAt: {
+    type: Date,
+    default: Date.now
+  }  
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+app.get('/thoughts', async (req, res) => {  
+  const allThoughts = await Thought.find().sort({ CreatedAt: -1 }).limit(20)
+  res.json(allThoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought({ message: req.body.message }).save()
+    res.json(newThought)
+  } catch (error) {    
+    res.status(400).json({ message: 'Invalid request', error })
+  } 
+})
+
+app.delete('/thoughts/:id', async (req, res) => {
```

Comment:
It's fun with a delete functionality but it was not supposed to be there according to the brief I think...